pkg/netns/netns_linux.go

```
@@ -168,19 +168,19 @@ func (h *NetNS) Do(f func() error) error {
 // If the netns of the current thread has not changed, this is effectively a no-op.
 func saveRestoreCurrentThreadNetNS() (func(), error) {
 	runtime.LockOSThread()
-	hostNSFD, err := getCurrent()
+	orig, err := getCurrent()
 	if err != nil {
 		// flaking out early, so unlock current thread
 		runtime.UnlockOSThread()
 		return nil, fmt.Errorf("get current namespace: %w", err)
 	}
 
 	return func() {
-		_ = unix.Setns(hostNSFD, unix.CLONE_NEWNET)
+		_ = unix.Setns(orig, unix.CLONE_NEWNET)
```

This is a crucial mistake. We can't ignore this error. If setting the thread back to its original netns fails, we should never call UnlockOSThread, as this will release the thread back into the schedulable pool.

---

Entirely correct, ugh. I wish I could reasonably fence this with unit tests. Well, at least I'm failing in public :D